### PR TITLE
[App Check] Implement FirebaseAppCheck as a shim of AppCheckCore

### PR DESCRIPTION
**Pull request by andrewheard**
Implemented `FirebaseAppCheck` as a shim / wrapper of the new [`AppCheckCore`](https://github.com/google/app-check) SDK.

#no-changelog

### CODE REVIEW

File: FirebaseCore.podspec [hunk 1]
```diff
@@ -47,8 +47,8 @@ Firebase Core includes FIRApp and FIROptions which provide central configuration
   s.watchos.framework = 'WatchKit'
 
   # Remember to also update version in `cmake/external/GoogleUtilities.cmake`
```
Comment (paulb777) [inline FirebaseCore.podspec:49]: On second thought, I think it would be safer to do this update since Core is going from 7.8 to 7.12

**Comment by andrewheard** [comment]
Note: The `FIRAppCheckProvider` public API change is already in `master` (https://github.com/firebase/firebase-ios-sdk/blob/4ff62df506a186f6d12bcacf33bc8b460c44a174/FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheckProvider.h#L43C1-L45C52).

File: FirebaseCore.podspec (hunk 1 above)
Comment (andrewheard) [reply FirebaseCore.podspec:49]: Agreed. Running CI in https://github.com/firebase/firebase-ios-sdk/pull/12069 now.

**Review by paulb777 — APPROVED** [review]
OK to merge and let the zip run in the nightlies

**Pull request merged**